snapshot.sh

```diff
@@ -31,9 +32,15 @@ while true; do
         aws_args="--endpoint-url ${S3_HOST}"
         s3_uri_base="s3://${SNAPSHOT_PATH}"
         timestamp=$(date +"%Y-%m-%dT%H:%M:%S")
-        s3_uri="${s3_uri_base}/${SNAPSHOT_PREFIX}_${timestamp}.tar.gz"
+        s3_uri="${s3_uri_base}/${SNAPSHOT_PREFIX}_${timestamp}.${SNAPSHOT_FORMAT}"
 
-        tar c -C $SNAPSHOT_DIR . | gzip | aws $aws_args s3 cp - "$s3_uri" --expected-size $SNAPSHOT_SIZE
+        SNAPSHOT_SIZE=$(du -sb $SNAPSHOT_DIR | cut -f1)
+        SNAPSHOT_COMPRESS=${SNAPSHOT_COMPRESS:-1}
+        if [ "$SNAPSHOT_COMPRESS" != "0" ]; then
+            (tar c -C $SNAPSHOT_DIR . | gzip -1 | pv -petrafb -i 5 -s $SNAPSHOT_SIZE | aws $aws_args s3 cp - "$s3_uri" --expected-size $SNAPSHOT_SIZE) 2>&1 | stdbuf -o0 tr '\r' '\n'
+        else
+            (tar c -C $SNAPSHOT_DIR . | pv -petrafb -i 5 -s $SNAPSHOT_SIZE | aws $aws_args s3 cp - "$s3_uri" --expected-size $SNAPSHOT_SIZE) 2>&1 | stdbuf -o0 tr '\r' '\n'
```

Comment:
Isn't it strange that if compression is disabled and the URL still defaults to `.tar.gz`?

Reply:
Fixed 2 days ago in the 2nd commit of this PR
https://github.com/ovrclk/cosmos-omnibus/pull/262/commits/f4ffd3f326e7d36fa1321edeaf82f1c6f953dd8c